server/room-battle.js

```diff
@@ -997,6 +997,14 @@ class RoomBattle extends RoomGames.RoomGame {
 		return player;
 	}
 
+	forcedPublic() {
+		if (!this.rated) return;
+		for (const userid in this.playerTable) {
```

Comment:
Ideally don't iterate `playerTable` ever.

`for (const player of this.players) {` should be what you're looking for.

Reply:
Don't they not have a user id though?

Reply:
They do.

```suggestion
		for (const player of this.players) {
```